### PR TITLE
Add instructions to DEVELOPMENT.md for installing tools

### PR DESCRIPTION
**Pull request by tejal29**
This commit, adds missing instructions to DEVELOPMENT.md like
- tools set up
- how to fix go-critic error

Also fixes bunch of whitespace eol
```
git diff --ignore-space-at-eol

+++ b/DEVELOPMENT.md
@@ -3,6 +3,27 @@
 This doc explains the development workflow so you can get started
 [contributing](CONTRIBUTING.md) to Skaffold!
 
+
+## Requirements
+
+You must install these tools:
+
+1. [`go`](https://golang.org/doc/install): The language skaffold is
+   built in
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [`dep`](https://github.com/golang/dep): For managing external Go
+   dependencies. - Please Install dep v0.5.0 or greater.
+1. [`make`](https://www.gnu.org/software/make/): For building skaffold.
+1. [`golangci-lint`](https://github.com/golangci/golangci-lint): You can use the
+   helper [script](./hack/install_golint.sh) to get golangci-lint.
+
+  ```shell
+    ./hack/install_golint.sh
+  ```
+
+1. [`gocritic`](https://github.com/go-critic/go-critic): Go source code linter
+   providing advanced checks currently missing from other linters.
+
 ## Getting started
 
 First you will need to setup your GitHub account and create a fork:
@@ -96,6 +117,16 @@ make test
 
 _These tests will not run correctly unless you have [checked out your fork into your `$GOPATH`](#checkout-your-fork)._
 
+In case you see go-critic tool error,
+
+```shell
+make test
+RUN hack/linter.sh
+ERRO Running error: no such linter "gocritic"
+```
+
+re-run the `hack/install_golint.sh` script to upgrade `golangci-lint`.
+
 ### Integration tests

```

### CODE REVIEW

**Comment by balopat** [comment]
Added this as a flake: https://github.com/GoogleContainerTools/skaffold/issues/1767

File: DEVELOPMENT.md [hunk 1]
```diff
@@ -3,6 +3,27 @@
...
+1. [`dep`](https://github.com/golang/dep): For managing external Go
+   dependencies. - Please Install dep v0.5.0 or greater.
+1. [`make`](https://www.gnu.org/software/make/): For building skaffold.
+1. [`golangci-lint`](https://github.com/golangci/golangci-lint): You can use the
+   helper [script](./hack/install_golint.sh) to get golangci-lint.
+
+  ```shell
+    ./hack/install_golint.sh
+  ```
+
```
Comment (balopat) [inline DEVELOPMENT.md]: remove new line? the next line restarts from 1

File: DEVELOPMENT.md [hunk 2]
```diff
@@ -3,6 +3,27 @@
...
+1. [`go`](https://golang.org/doc/install): The language skaffold is
+   built in
+1. [`git`](https://help.github.com/articles/set-up-git/): For source control
+1. [`dep`](https://github.com/golang/dep): For managing external Go
+   dependencies. - Please Install dep v0.5.0 or greater.
+1. [`make`](https://www.gnu.org/software/make/): For building skaffold.
+1. [`golangci-lint`](https://github.com/golangci/golangci-lint): You can use the
+   helper [script](./hack/install_golint.sh) to get golangci-lint.
+
```
Comment (balopat) [inline DEVELOPMENT.md]: remove new line? to move the 6 th point to the same block?

Comment (tejal29) [reply DEVELOPMENT.md]: the mdlint rule mentions empty line before and after "shell" block.

**Comment by dgageot** [comment]
@balopat @tejal29 is this PR good to go?

**Comment by nkubala** [comment]
kokoro is drunk again, restarting the build. once this is green LGTM

File: DEVELOPMENT.md (hunk 2 above)
Comment (tejal29) [reply DEVELOPMENT.md]: will fix it.

**Pull request merged**